src/pveclib/vec_f128_ppc.h

```diff
@@ -1184,13 +1377,162 @@ vec_signbitf128 (__binary128 f128)
 #if defined (_ARCH_PWR9) && defined (scalar_test_neg) && (__GNUC > 7)
   return (scalar_test_neg (f128);
 #else
-  vui32_t tmp, t128;
+  vui32_t tmp;
   const vui32_t signmask = CONST_VINT128_W(0x80000000, 0, 0, 0);
 
-  t128 = vec_xfer_bin128_2_vui32t (f128);
-  tmp = vec_and (t128, signmask);
+  tmp = vec_and_bin128_2_vui32t (f128, signmask);
   return vec_all_eq(tmp, signmask);
 #endif
 }
 
+/** \brief Scalar Insert Exponent Quad-Precision
+ *
+ *  Merge the sign (bit 0) and significand (bits 16:127) from sig
+ *  with the 15-bit exponent from exp (bits 49:63). The exponent is
+ *  moved to bits 1:15 of the final result.
+ *  The result is return as a Quad_precision floating point value.
+ *
+ *  \note This operation is equivalent to the POWER9 xsiexpqp
+ *  instruction and the built-in scalar_insert_exp. These require a
+ *  POWER9 enables compiler targeting -mcpu=power9 and are not
+ *  available for older compilers and POWER8 and earlier.
+ *  This operation provides this operation for all VSX enabled
+ *  platforms.
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   |  2-11 | 2/cycle  |
+ *  |power9   |   2   | 4/cycle  |
+ *
+ *  @param sig Vector __int128 containing the Sign Bit and 112-bit significand.
+ *  @param exp Vector long long containing the 15-bit exponent.
+ *  @return a __binary128 value where the exponent bits (1:15) of sig
+ *  are replaced from bits 49:63 of exp.
+ *
+ */
+static inline __binary128
+vec_xsiexpqp (vui128_t sig, vui64_t exp)
+{
+  __binary128 result;
+#if defined (_ARCH_PWR9) && defined (__FLOAT128__) && (__GNUC__ > 7)
+  __asm__(
+      "xsiexpqp %0,%1,%2"
+      : "=v" (result)
+      : "v" (sig), "v" (exp)
+      : );
+
+#else
+  vui32_t tmp, t128;
+  const vui32_t expmask = CONST_VINT128_W(0x7fff0000, 0, 0, 0);
+
+  tmp = vec_sld ((vui32_t) exp, (vui32_t) exp, 6);
+  t128 =  vec_sel ((vui32_t) sig, tmp, expmask);
+  result = vec_xfer_vui32t_2_bin128 (t128);
+#endif
+  return result;
+}
+
+/** \brief Scalar Extract Exponent Quad-Precision
+ *
+ *  Extract the quad-precision exponent (bits 1:15) and right justify
+ *  it to (bits 49:63 of) doubleword 0 of the result vector.
+ *  The result is return as vector long long integer value.
+ *
+ *  \note This operation is equivalent to the POWER9 xsxexpqp
+ *  instruction and the built-in scalar_extract_exp. These require a
+ *  POWER9 enables compiler targeting -mcpu=power9 and are not
+ *  available for older compilers and POWER8 and earlier.
+ *  This operation provides this operation for all VSX enabled
```

Comment:
like above, s/POWER9 enables/POWER9-enabled/
s/and/nor/ on the 2nd line
s/VSX enabled/VSX-enabled/

Reply:
OK